lib/js/components/Pagination/Pagination.vue

```diff
@@ -0,0 +1,369 @@
+@import '../../../styles/settings/radiuses';
+@import '../../../styles/mixins/layout';
+
+$pagination-item-width: 32px;
+$pagination-item-height: 32px;
+
+.pagination {
+	$self: &;
+
+	@include centeredSpread();
+
+	flex-grow: 1;
+	flex-direction: row;
+	align-items: stretch;
+
+	&__itemsWrapper {
+		display: flex;
+		flex-wrap: nowrap;
+		align-content: center;
+		justify-content: center;
+	}
+
+	&.-compact {
+		#{$self}__items.-compact {
+			display: flex !important;
+		}
+
+		#{$self}__items.-default {
+			display: none;
+		}
+	}
+
+	&.-centered {
+		#{$self}__itemsWrapper {
+			flex-grow: 1;
+			align-items: center;
+		}
+	}
+
+	&__items {
+		display: none;
+		flex-direction: row;
+		align-items: center;
+		padding: 0;
+
+		@media #{breakpoint-s()} {
+			&.-default {
+				display: flex;
+			}
+
+			&.-compact {
+				display: none !important;
+			}
+		}
+
+		&.-compact {
+			display: flex;
+		}
+	}
+
+	&__input {
+		padding: $space-xxxs $space-xxs;
+		background: $color-default-background;
+		border: 1px solid $color-neutral-border-strong;
+		box-shadow: inset 0 1px 3px $color-neutral-border-strong;
+		border-radius: $radius-s;
+		height: $pagination-item-height;
+
+		&:focus {
+			outline: none;
+		}
+	}
+
+	&__text {
+		@include textM();
+
+		text-align: center;
+		color: $color-neutral-text;
+		padding: $space-xxxs;
+	}
+
+	&__compactItem {
+		@include centeredSpread();
+
+		flex-direction: row;
+	}
+
+	&__itemWrapper {
+		@include textM();
+
+		text-align: center;
+		padding: $space-xxxxs;
+	}
+
+	&__item {
+		display: flex;
+		flex-direction: column;
+		justify-content: center;
+		align-items: center;
+		width: $pagination-item-width;
+		height: $pagination-item-height;
+		background: $color-neutral-background-weak;
+		border-radius: $radius-xs;
+		color: $color-neutral-text;
+		text-decoration: none;
+
+		&.-selected {
+			@include textBold();
+
+			background: $color-neutral-background-medium;
+			color: $color-neutral-text-heavy;
+		}
+
+		&:active,
+		&:focus,
+		&:hover {
+			outline: none;
+		}
+
+		&:hover:not(.-selected) {
+			background: $color-neutral-background-weak-hovered;
+			cursor: pointer;
+		}
+
+		&:active:not(.-selected) {
+			background: $color-neutral-background-weak-hovered;
+			box-shadow: $shadow-inset-s;
+		}
+	}
+
+	&__ellipsis {
+		display: flex;
+		flex-direction: column;
+		justify-content: center;
+		align-items: center;
+		width: 22px;
+		height: $pagination-item-height;
+		color: $color-neutral-text-weak;
+	}
+
+	&__accessorySlot {
+		display: flex;
+		align-items: center;
+		padding: 0;
+		min-height: 0;
+	}
+}
+</style>
+
+<script lang="ts">
+import { PAGINATION_DEFAULT_ITEMS_PER_PAGE, PAGINATION_LAYOUTS } from './Pagination.consts';
+import IconButton from '../Buttons/IconButton/IconButton.vue';
+import { ICON_BUTTON_COLORS, ICON_BUTTON_SIZES, ICON_BUTTON_STATES } from '../Buttons/IconButton';
+import { ICONS } from '../Icon';
+
+const MAX_DISPLAYED_ITEMS = 7;
```

Comment:
The name Items occurs in props and have other meaning. I'd change it for navigation elements